### PR TITLE
Project restructuring

### PR DESCRIPTION
**Pull request by chivay**

### CODE REVIEW

**Comment by msm-code** [comment]
Files changed: 216

Ah yes, enslaved repository

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,70 +1,34 @@
...
-    RawFile.cpp
-    OnDiskFileIndex.cpp
-)
-target_include_directories(ursa PRIVATE ${PROJECT_BINARY_DIR})
-target_link_libraries(ursa -lstdc++fs -lzmq -pthread)
+add_subdirectory(libursa)
 
-add_executable(ursadb Daemon.cpp NetworkService.cpp)
+add_executable(ursadb src/Daemon.cpp
+                      src/Daemon.h
```
Comment (msm-code) [inline CMakeLists.txt:14]: Should we add headers too? AFAIK usually X.cpp should always include X.h anyway.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,70 +1,34 @@
...
-    Responses.cpp
-    RawFile.cpp
-    OnDiskFileIndex.cpp
-)
-target_include_directories(ursa PRIVATE ${PROJECT_BINARY_DIR})
-target_link_libraries(ursa -lstdc++fs -lzmq -pthread)
+add_subdirectory(libursa)
 
-add_executable(ursadb Daemon.cpp NetworkService.cpp)
+add_executable(ursadb src/Daemon.cpp
```
Comment (msm-code) [inline CMakeLists.txt:13]: nitpick: not a fan of aligning. Is it a standard in cmake? If no, maybe continue from newline with 4 spaces indent (like in libursa)?

File: libursa/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    Database.cpp
+    DatasetBuilder.cpp
+    FlatIndexBuilder.h
+    MemMap.h
+    OnDiskIndex.h
+    QueryParser.h
+    Task.cpp
+    ZHelpers.h
+)
+configure_file(Version.h.in
```
Comment (msm-code) [inline libursa/CMakeLists.txt:51]: newline after `(` or all in single line, maybe?

File: CMakeLists.txt (hunk 1 above)
Comment (chivay) [reply CMakeLists.txt:14]: Actually, it's intended. CMake is smart and can figure it out itself, however it's meant for clangformat target.

From a few resources, I've found out that that globbing is a bad practice in cmake and adding them here explicitly will make those files discoverable for clang-format.

File: CMakeLists.txt (hunk 2 above)
Comment (chivay) [reply CMakeLists.txt:13]: sure.

**Pull request merged**